src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -105,25 +103,26 @@ public ConsumeService(Map<String, Object> props, String name) throws Exception {
 
     // Assign config specified for consumer. This has the highest priority.
     consumerProps.putAll(consumerPropsOverride);
-
     _consumer = (KMBaseConsumer) Class.forName(consumerClassName).getConstructor(String.class, Properties.class).newInstance(topic, consumerProps);
-    _thread = new Thread(() -> {
-      try {
-        consume();
-      } catch (Exception e) {
-        LOG.error(_name + "/ConsumeService failed", e);
-      }
-    }, _name + " consume-service");
-    _thread.setDaemon(true);
-
-    MetricConfig metricConfig = new MetricConfig().samples(60).timeWindow(1000, TimeUnit.MILLISECONDS);
-    List<MetricsReporter> reporters = new ArrayList<>();
-    reporters.add(new JmxReporter(JMX_PREFIX));
-    Metrics metrics = new Metrics(metricConfig, reporters, new SystemTime());
-    Map<String, String> tags = new HashMap<>();
-    tags.put("name", _name);
-    _adminClient = AdminClient.create(props);
-    _sensors = new ConsumeMetrics(metrics, tags, topic);
+    TopicManagementService.topicPartitionReady().thenRun(() -> {
+      MetricConfig metricConfig = new MetricConfig().samples(60).timeWindow(1000, TimeUnit.MILLISECONDS);
+      List<MetricsReporter> reporters = new ArrayList<>();
+      reporters.add(new JmxReporter(JMX_PREFIX));
+      Metrics metrics = new Metrics(metricConfig, reporters, new SystemTime());
+      Map<String, String> tags = new HashMap<>();
+      tags.put("name", _name);
+      _adminClient = AdminClient.create(props);
+      _sensors = new ConsumeMetrics(metrics, tags, topic);
+      _thread = new Thread(() -> {
+        try {
+          consume();
+        } catch (Exception e) {
```

Comment:
This approach, waiting for the topic to be ready, seems like it will fix some problems with order.  But making the consume service more resilient to error conditions in general will handle cases that has not yet been thought of.  When is it OK to exit the consume service?

Reply:
Updated the PR. Which other avenues to render consume service more resilient to error conditions do you suggest? Currently, with the combinations of `asynchronous` calls due to the nature of Kafka Monitor since its inception and the usages of the new `AdminClient` API, it is appropriate to align the executions.